src/InsertionsClient.Console/ConsoleApp/InputLoading.cs

```diff
@@ -111,5 +113,185 @@ internal static List<string> LoadManifestPaths(string manifestPaths, out int inv
+                        {
+                            // We haven't put any code into the expression of this ruleset. This will be the first.
+                            rulesetExpression = ruleExpression;
+                        }
+                        else
+                        {
+                            // This ruleset already contains some code from the previous rules. Combine them with "And".
+                            rulesetExpression = Expression.AndAlso(rulesetExpression, ruleExpression);
+                        }
+                    }
+
+                    if(rootExpression == null)
+                    {
+                        // We haven't put any code into the root expression. This will be the first.
+                        rootExpression = rulesetExpression;
+                    }
+                    else
+                    {
+                        // Root expression already contains some code from the previous rulesets. Combine the new one with "Or".
+                        rootExpression = Expression.OrElse(rootExpression, rulesetExpression);
+                    }
+
+                }
+
+                Expression<Predicate<Build>> lambda = Expression.Lambda<Predicate<Build>>(rootExpression, false, buildParameter);
+                buildFilter = lambda.Compile();
+                return true;
+            }
+            catch (Exception e)
+            {
+                Trace.WriteLine($"Parsing the build filter has failed unexpectedly. Exception: {e.ToString()}");
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Creates a predicate that checks if the given property matches the provided regular expression.
+        /// </summary>
+        /// <param name="propertyGetter">Getter that will return a string property from a <see cref="Build"/></param>
+        /// <param name="regex">Regular expression to be used for evaluating the value of the property.</param>
+        /// <returns>The created predicate.</returns>
+        internal static Predicate<Build> CreateBuildPredicateBasedOnStringProperty(Func<Build, string?> propertyGetter, Regex regex)
+        {
+            return (build) =>
+            {
+                string? value = propertyGetter(build);
+                Match? match = regex.Match(value);
+
+                if (match == null || !match.Success)
+                {
+                    return false;
+                }
+
+                return match.Length == (value?.Length ?? 0);
+            };
+        }
+
+        /// <summary>
+        /// Creates a predicate that checks if the provided regex matches any of the channels of the given build.
+        /// </summary>
+        /// <param name="regex">Regular expression to be used for evaluating the names of build channels.</param>
+        /// <returns>The created predicate.</returns>
+        internal static Predicate<Build> CreateBuildPredicateBasedOnChannel(Regex regex)
```

Comment:
consider maintaining all "Create..." methods in a distinct expression-factory class.